### DISCUSSION TITLE
Dynamically control max charging power based on total consumption from the mains

### DISCUSSION THREAD

**Discussion by devdems** [serkri/SmartEVSE-3#243] [Q&A] [answered]
What would be the best way to dynamically control SmartEVSE charging power based on the feed information from the mains consumption meter?

We have regulations where the total electricity consumption throughout the day varies. 

For example from 7:00 - 13:00 total consumption can be 8kw, 13:00 - 14:00 10kw, 14:00 - 20:00 20kw...

How can I limit charging so that the total consumption will not exceed what is allowed? I already have the logic built in a script and was thinking of controlling the SmartEVSE trough API or MQTT. What is the setting I need to be changing?

**Comment by dingo35** [comment]
EDIT: removed comment

**Comment by dingo35** [comment]
EDIT: removed comment

**Comment by devdems** [comment]


**Reply by dingo35** [reply]
Excuse me I thought this showed up in my Issues timeline. Again my apologies.

**Comment by dingo35** [comment]
To answer your question, no need to change any settings, you can just set OverrideCurrent via both REST API or MQTT.
Homeassistant would be an easy way to keep track of your usage and calculate what charge current would be appropriate for the rest of the time period you are in.

**Comment by dingo35** [comment]
You can find docs here:
https://github.com/dingo35/SmartEVSE-3.5/blob/master/docs/configuration.md

**Comment by devdems** [comment]
Isn't setting OverrideCurrent just setting the max charge current for the charger? Then this means I would need to calculate the max charge current with the script and push this value to the charge which I can do.

But I was thinking of a more global setting where you can set **max grid consumption** and then the charger figures out max charge current based on this setting and based on the current power consumption. So it will still be dynamic.

Hope this make sense.

**Comment by dingo35** [comment, marked as answer]
Look SumMains, that comes closest to your use case.

The SmartEVSE does not keep track of total energy used, it can only report those values from an external source like a kWh meter.

Also, you should read up on your physics: if you want to limit power, unit is W or kW. If you want to limit energy, unit is Wh or kWh.

Either way your use case is so specific that I am not putting this in mainstream code. But you can always make your own changes, its open source!

**Comment by devdems** [comment]
Great, tnx for the answer! This is exactly what I was looking for. We have similar regulations then in Belgium for which I see that this feature was done: https://github.com/serkri/SmartEVSE-3/issues/215
